libraries/AP_Airspeed/AP_Airspeed.cpp

```diff
@@ -399,16 +408,90 @@ void AP_Airspeed::read(void)
     }
 #endif
 
-    // setup primary
-    if (healthy(primary_sensor.get())) {
-        primary = primary_sensor.get();
-        return;
-    }
-    for (uint8_t i=0; i<AIRSPEED_MAX_SENSORS; i++) {
-        if (healthy(i)) {
-            primary = i;
-            break;
+    switch(blend_conf) {
+    case BLEND_NONE:
+        // setup primary
+        if (healthy(primary_sensor.get())) {
+            primary = primary_sensor.get();
+        } else {
+            for (uint8_t i=0; i<AIRSPEED_MAX_SENSORS; i++) {
+                if (healthy(i)) {
+                    primary = i;
+                    break;
+                }
+            }
+        }
+        break;
+    case BLEND_FULL:
+        // TODO: check have_inertial_nav()?
+
+        // TODO: check _ahrs != NULL
+
+        // blend the airspeed from several sensors and EKF
+        Vector3f velocity, wind;
+        if (_ahrs->get_velocity_NED(velocity) == false) {
+            // TODO: correct calculation
+            return;
         }
+        wind = _ahrs->wind_estimate();
+        velocity -= wind;
+        float airspeed_ekf = velocity.length();
+
+        float airspeed_avg = 0.0;
+        float ekf_sensor_airspeed_diff[AIRSPEED_MAX_SENSORS];
+        float avg_sensor_airspeed_diff[AIRSPEED_MAX_SENSORS];
+        for (uint8_t i = 0; i < AIRSPEED_MAX_SENSORS; i++) {
+            airspeed_avg += state[i].airspeed;
+        }
+        airspeed_avg /= AIRSPEED_MAX_SENSORS;
+
+        // calculating inverse square of error that will be our trust in sensor
+        for (uint8_t i = 0; i < AIRSPEED_MAX_SENSORS; i++) {
+            ekf_sensor_airspeed_diff[i] = state[i].airspeed - airspeed_ekf;
+            ekf_sensor_airspeed_diff[i] = 1 / (ekf_sensor_airspeed_diff[i] * ekf_sensor_airspeed_diff[i]);
+            avg_sensor_airspeed_diff[i] = state[i].airspeed - airspeed_avg;
+            avg_sensor_airspeed_diff[i] = 1 / (avg_sensor_airspeed_diff[i] * avg_sensor_airspeed_diff[i]);
+        }
+
+        // Main idea is that we trust more to sensors that are close together (if there are >= 3)
```

Comment:
If you have three or more sensors, the most reliable algorithm is to reject the two outliers and take the mean of the centre value(s). Using the EKF prediction to select weightings is an unnecessary complexity that introduces circular data dependency.

Reply:
Right now we are limited to 2 and in 99% of cases 2 sensors will be the real maximum number.
I worried about dependency introduced by such method too and I would really like to hear your thoughts on this in case when we have 2 sensors.